datamodel.itop-communications.xml

```diff
@@ -232,33 +244,40 @@
 	 */]]></comment>
           <code><![CDATA[    public function IsUserInScope(User $oUser)
     {
-		$oOrgSearch = DBSearch::FromOQL_AllData('SELECT lnkCommunicationToOrganization WHERE communication_id = :communication_id');
-		$oOrgSet = new DBObjectSet($oOrgSearch, array(), array('communication_id' => $this->GetKey()));
-		$aTargetOrgs = $oOrgSet->GetColumnAsArray('org_id');
-		if (count($aTargetOrgs) == 0)
-		{
-			$bRet = true;
-		}
-		else
-		{
-			$iContact = $oUser->Get('contactid');
-			$oContact = MetaModel::GetObject('Contact', $iContact, true, true);
-			$iUserOrg = $oContact->Get('org_id');
+		// Is the user org within the target organizations
+		$iContact = $oUser->Get('contactid');
+		$oContact = MetaModel::GetObject('Contact', $iContact, true, true);
+		$iUserOrg = $oContact->Get('org_id');
+		$sMatchType = $this->Get('org_match_type');
+        $bRet = false;
 
-			$sMatchType = $this->Get('org_match_type');
-			if ($sMatchType == 'direct')
-			{
-				// Flat version
-				$bRet = in_array($iUserOrg, $aTargetOrgs);
-			}
-			else // 'cascade'
-			{
-				// Is the user org below or within any of the target organizations
-				$oOrgHierarchical = DBSearch::FromOQL_AllData('SELECT Organization AS child JOIN Organization AS root ON child.parent_id BELOW root.id WHERE root.id IN (:target_org_list) AND child.id = :contact_org_id');
-				$oOrgHSet = new DBObjectSet($oOrgHierarchical, array(), array('contact_org_id' => $iUserOrg, 'target_org_list' => $aTargetOrgs));
-				$bRet = $oOrgHSet->Count() > 0;
-			}
+		if ($sMatchType == 'oql') {
```

Comment:
```suggestion
		if ($sMatchType === 'oql') {
```

We tend to prefer strict comparison when possible (= variable type is known to alsays be the same), this avoid regression in the future.